#!/bin/bash

#scRepli-seq pipeline
#Step3 load mapped reads in Rdata format by AneuFinder
#This script will generate ${out_dir}/bins, ${out_dir}/fragment and save the Rdata 

if [ $# -le 4 ] ; then
    echo "Usage: bash Step3_load_mapped_reads.sh [bam_file] [out_dir] [name] [blacklist] [genome_file]"
    echo ""
    echo "Load the mapped reads, convert into Rdata format, and binning the data in Rdata format"
    echo ""
    echo "arguments:"
    echo ""
    echo "bam_file			Input bam file (generated by Step2 mapping)"
    echo "out_dir			Output directory (generating bins & fragment folder)"
    echo "name				Output file name"
    echo "blacklist			black list file ex) ENCFF000KJP_name2id.bed etc.."
    echo "genome_file		genome file ex) hg19_female.fa.fai, mm9.chrom.size etc.."
    echo ""
    echo "Output file name will be"
    echo "[name]_mapq10_blacklist_fragment.Rdata"
    echo "[name]_mapq10_blacklist_bin.Rdata"
    echo "BinSizes are 40000,80000,100000,200000,500000 bps"
    echo ""
    exit 0
fi

echo "#scRepli-seq Step3 load mapped reads start: `date`"

rscript="/usr/local/bin/util/Step3_R-Aneu-Fragment-bins.R"
bamfile=$1
out_dir=$2
name=$3 
blacklist=$4
genome_file=$5

filename=`basename $bamfile`
echo "#file name:${filename}"


Rscript --vanilla $rscript ${bamfile} ${out_dir} ${name} ${blacklist} ${genome_file}

echo "#scRepli-seq Step3 load mapped reads end: `date`"


